airflow/utils/log/file_processor_handler.py

```diff
@@ -54,6 +54,8 @@ def set_context(self, filename):
         :param filename: filename in which the dag is located
         """
         local_loc = self._init_file(filename)
+        if self.handler:
+            self.handler.close()
```

Comment:
I think we can you the method on Line 75
```suggestion
         self.close()
```

Reply:
Nope. That would be wrong. The whole idea is that the main handler is not closed, we should only close the underlying handler without actually closing the main one.